### PR TITLE
[SPARK-23122][PYSPARK][FOLLOW-UP] Update the docs for UDF Registration

### PR DESCRIPTION
**Pull request by gatorsmile**
## What changes were proposed in this pull request?

This PR is to update the docs for UDF registration  

## How was this patch tested?

N/A

### CODE REVIEW

File: python/pyspark/sql/udf.py [hunk 1]
```diff
@@ -200,7 +200,7 @@ def __init__(self, sparkSession):
     @since("1.3.1")
     def register(self, name, f, returnType=None):
         """Registers a Python function (including lambda function) or a user-defined function
```
Comment (ueshin) [inline python/pyspark/sql/udf.py]: `Register` instead of `Registers` to be consistent with other descriptions?

**Review by ueshin — COMMENTED** [review]
LGTM except for one comment.

**Review by HyukjinKwon — APPROVED** [review]
LGTM. @gatorsmile, sorry I don't know why I missed your comment in https://github.com/apache/spark/pull/20288.

File: python/pyspark/sql/udf.py [hunk 2]
```diff
@@ -213,6 +213,10 @@ def register(self, name, f, returnType=None):
         `returnType` can be optionally specified when `f` is a Python function but not
         when `f` is a user-defined function. Please see below.
 
+        To register a non-deterministic Python function, users need to first build
+        a nondeterministic user-defined function for the Python function and then register it
```
[the remark this answers is not in the document]
Comment (gatorsmile) [reply python/pyspark/sql/udf.py]: `nondeterministic` is better

**Comment by gatorsmile** [comment]
@HyukjinKwon That is fine. I am reviewing all the API changes made in Spark 2.3 release. 

Thanks! Merged to master/2.3

**Pull request closed without merging**